SingularityService/src/main/java/com/hubspot/singularity/data/StateManager.java

```diff
@@ -195,6 +198,7 @@ public SingularityState generateState(boolean includeRequestIds) {
       updatePossiblyUnderProvisionedAndOverProvisionedIds(requestWithState, numInstances, overProvisionedRequestIds, possiblyUnderProvisionedRequestIds);
     }
 
+    removePendingRequestIds(possiblyUnderProvisionedRequestIds);
```

Comment:
can we give this a better name? Like `filterForPendingRequests` or something more descriptive like that?